patches/server/0807-Allow-explosions-to-affect-projectiles-v2.patch

```diff
@@ -0,0 +1,37 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: c0repwn3r <coredoescode@gmail.com>
+Date: Wed, 22 Sep 2021 15:26:37 -0400
+Subject: [PATCH] Allow explosions to affect projectiles v2
+
+
+diff --git a/src/main/java/net/minecraft/world/level/Explosion.java b/src/main/java/net/minecraft/world/level/Explosion.java
+index 132140e00865fcf84ebe03ffcbc2f30ac11a0b35..11739e6ec31fe2f11d0ce8a645a000572707de5d 100644
+--- a/src/main/java/net/minecraft/world/level/Explosion.java
++++ b/src/main/java/net/minecraft/world/level/Explosion.java
+@@ -30,6 +30,9 @@ import net.minecraft.world.entity.item.ItemEntity;
+ import net.minecraft.world.entity.item.PrimedTnt;
+ import net.minecraft.world.entity.player.Player;
+ import net.minecraft.world.entity.projectile.Projectile;
++import net.minecraft.world.entity.projectile.Fireball;
++import net.minecraft.world.entity.projectile.DragonFireball;
++import net.minecraft.world.entity.projectile.SmallFireball;
+ import net.minecraft.world.item.ItemStack;
+ import net.minecraft.world.item.enchantment.ProtectionEnchantment;
+ import net.minecraft.world.level.block.BaseFireBlock;
+@@ -249,9 +252,15 @@ public class Explosion {
+                         entity.forceExplosionKnockback = false;
+                         boolean wasDamaged = entity.hurt(this.getDamageSource(), (float) ((int) ((d13 * d13 + d13) / 2.0D * 7.0D * (double) f2 + 1.0D)));
+                         CraftEventFactory.entityDamage = null;
+-                        if (!wasDamaged && !(entity instanceof PrimedTnt || entity instanceof FallingBlockEntity) && !entity.forceExplosionKnockback) {
++			// Paper start - allow explosions to affect projectiles except for Fireballs
++			boolean isFireball = (entity instanceof Fireball || entity instanceof DragonFireball || entity instanceof SmallFireball);
++                        if (!wasDamaged &&
++					!(entity instanceof PrimedTnt ||
++					  entity instanceof FallingBlockEntity ||
++					 (entity instanceof Projectile && !isFireball)) && !entity.forceExplosionKnockback) {
```

Comment:
Couple things here:
1. you are using tabs instead of spaces
2. This is indented all over the place. I'd just keep it all on the same line, or read up on the conventions for multi-line conditionals.
3. The `//Paper end` comment is too far down. It should be right after the last changed line.

Reply:
Ok, let fix that up real quick.